content/cucumber/hooks.md

```diff
@@ -76,71 +148,136 @@ end
+{{% text "javascript" %}}`Before` or `After`{{% /text %}}
+Hook with one or more [Tags](/cucumber/tags/).
 
-Pass `OR` Tags in a single string, comma-separated:
+You can use OR and AND and NOT Tags in much the same way as you can when running Cucumber from the command line. Examples:
 
 ```ruby
+# Pass `OR` Tags in a single string, comma-separated:
 Before('@cucumis, @sativus') do
   # This will only run before scenarios tagged
   # with @cucumis OR @sativus.
 end
-```
-
-Pass `AND` Tags as separate strings:
 
-```ruby
+# Pass `AND` Tags as separate strings:
 Before('@cucumis', '~@sativus') do
   # This will only run before scenarios tagged
   # with @cucumis AND NOT @sativus.
-end
-```
 
-You create complex Tag conditions using both `OR` and `AND` on Tags:
-
-```ruby
+# You create complex Tag conditions using both `OR` and `AND` on Tags:
 Before('@cucumis, @sativus', '@aqua') do
   # This will only run before scenarios tagged
   # with (@cucumis OR @sativus) AND @aqua
 end
-```
-
-`AfterStep` example:
 
-```ruby
+# `AfterStep` example:
 AfterStep('@cucumis', '@sativus') do
   # This will only run after steps within scenarios tagged
   # with @cucumis AND @sativus.
 end
+
+# Pass `NOT` Tag
+Before('NOT @cucumis') do
+  # This will only run before scenarios NOT tagged
+  # with @cucumis
+end
+```
+
+```java
+@Before
+public void beforeScenario(){
+    // This hook will be executed before all scenarios
+}
+
+@Before("@foo")
+public void beforeScenario(){
+    // This hook will be executed before scenarios tagged with @foo
+}
+
+@Before("@foo and @bar")
+public void beforeScenario(){
+    // This hook will be executed before scenarios tagged with @foo and @bar
+}
+
+@Before("@foo or @bar")
+public void beforeScenario(){
+    // This hook will be executed before scenarios tagged with @foo or @bar
+}
+
+@Before("not @foo")
+public void beforeScenario(){
+    // This hook will be executed before scenarios NOT tagged with @foo
+}
+```
+
+```javascript
+var {defineSupportCode} = require('cucumber');
+
+defineSupportCode(function({After, Before}) {
+  Before(function () {
+    // This hook will be executed before all scenarios
+  });
+
+  Before({tags: "@foo"}, function () {
+    // This hook will be executed before scenarios tagged with @foo
+  });
+
+  Before({tags: "@foo and @bar"}, function () {
+    // This hook will be executed before scenarios tagged with @foo and @bar
+  });
+
+  Before({tags: "@foo or @bar"}, function () {
+    // This hook will be executed before scenarios tagged with @foo or @bar
+  });
+
+  Before({tags: "not @foo"}, function () {
+    // This hook will be executed before scenarios NOT tagged with @foo
+  });
+
+  // You can use the following shorthand when only specifying tags
+  Before("@foo", function () {
+    // This hook will be executed before scenarios tagged with @foo
+  });
+});
 ```
 
+See more documentation on [tag expressions](https://docs.cucumber.io/tag-expressions/).
```

Comment:
Is this link meant to be a local relative link?

Reply:
Atm it's not; perhaps it should be? (i.e. the info on Tag Expressions should be added to these docs)

Reply:
This repository has source code for a Web site which is not currently officially live, is it?